cmd/kubeadm/app/phases/apiconfig/setupmaster.go

```diff
@@ -51,15 +53,16 @@ func attemptToUpdateMasterRoleLabelsAndTaints(client *clientset.Clientset) error
 		return err
 	}
 
-	// TODO: Switch to the new master label defined in https://github.com/kubernetes/kubernetes/pull/39112
-	n.ObjectMeta.Labels[metav1.NodeLabelKubeadmAlphaRole] = metav1.NodeLabelRoleMaster
-
 	// TODO: Use the Taints beta field on the NodeSpec now
 	taintsAnnotation, _ := json.Marshal([]v1.Taint{{Key: "dedicated", Value: "master", Effect: "NoSchedule"}})
-	n.ObjectMeta.Annotations[v1.TaintsAnnotationKey] = string(taintsAnnotation)
 
-	// TODO: Use a patch instead of an Update
-	if _, err := client.Nodes().Update(n); err != nil {
+	// TODO: Switch to the new master label defined in https://github.com/kubernetes/kubernetes/pull/39112
+	addAnnotationPatch := fmt.Sprintf(
```

Comment:
I wouldn't recommend building a patch by hand.  As I recall there are functions that take two objects and compute the diff.